snapcraft/internal/meta/_snap_packaging.py

```diff
@@ -129,7 +129,7 @@ def _update_yaml_with_extracted_metadata(
     config_data: Dict[str, Any],
     parts_config: project_loader.PartsConfig,
     prime_dir: str,
-) -> None:
+) -> extractors.ExtractedMetadata:
     if "adopt-info" in config_data:
```

Comment:
can we instead of `return None` at the very end, just add a check here,
```suggestion
    if "adopt-info" not in config_data:
        return None
```

And move everything beneath this one level up?

Reply:
Yes, it looks better. Changed this one, but I think meta has still many other places where it can be improved in a future refactoring.